### PR TITLE
Detect if browser prefers night mode

### PR DESCRIPTION
**Pull request by btkostner**
Detects if the browser prefers a dark color scheme and switches to dark mode automatically. Needs testing.

Relevant APIs:
https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme
https://developer.mozilla.org/en-US/docs/Web/API/Window/matchMedia

### CODE REVIEW

File: assets/js/night.js [hunk 1]
```diff
@@ -17,12 +17,14 @@ function activateNightMode () {
 
 function deactivateNightMode () {
   body.removeClass(nightMode)
-  try { localStorage.removeItem(nightMode) } catch (e) { }
+  try { localStorage.setItem(nightMode, false) } catch (e) { }
```
Comment (josevalim) [inline assets/js/night.js]: Is this change necessary or is it a style change?

Comment (btkostner) [reply assets/js/night.js]: It's necessary to tell if a user intentionally set the theme to day time. That way we can prefer what the user has set, and fall back to what the browser wants.

I just realized this still has a bug, will fix.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:
